pkg/transport/transport.go

```diff
@@ -30,24 +30,37 @@ type Client interface {
 	WithObservers(obs ...Observer) Client
 }
 
+// Timeouts contains all of the timeouts configurable for a transport
+type Timeouts struct {
+	Request time.Duration
+	Dial    time.Duration
+}
+
 // Transport interface structure
 type Transport struct {
-	tlsOpts        *tlsopts.Options
-	observers      []Observer
-	requestTimeout time.Duration
+	tlsOpts   *tlsopts.Options
+	observers []Observer
+	timeouts  Timeouts
 }
 
 // NewClient returns a transport client with a default timeout for requests
 func NewClient(tlsOpts *tlsopts.Options, obs ...Observer) Client {
-	return NewClientWithTimeout(tlsOpts, defaultRequestTimeout, obs...)
+	return NewClientWithTimeouts(tlsOpts, Timeouts{}, obs...)
```

Comment:
maybe Timeouts should have the defaults set instead of being empty, unless it happens elsewhere? otherwise won't it be set to 0?

Reply:
Yep, the defaults are set in the `NewClientWithTimeouts` function so that no one can accidentally specify 0.

Reply:
OH I see. okay great!